server/service/devices.go

```diff
@@ -304,6 +325,36 @@ func (svc *Service) AuthenticateDeviceByCertificate(ctx context.Context, certSer
 	return host, svc.debugEnabledForHost(ctx, host.ID), nil
 }
 
+// AuthenticateDeviceByURL returns the host identified by the URL UUID.
+// This is used for devices accessing endpoints via a unique URL parameter.
+// Returns an error if the UUID doesn't exist or if the host is not iOS/iPadOS.
+func (svc *Service) AuthenticateDeviceByURL(ctx context.Context, urlUUID string) (*fleet.Host, bool, error) {
```

Comment:
Maybe `AuthenticateMobileAppleByURL`? Not a strong enough opinion to force a refactor on this if I didn't have other changes, but we can bake the whole iOS/iPadOS bit into the middleware name for clarity.

Reply:
Yes, much better. Done.